TYPO3.Flow/Classes/TYPO3/Flow/Security/Authentication/AuthenticationProviderManager.php

```diff
@@ -318,11 +319,22 @@ protected function buildProvidersAndTokensFromConfiguration(array $providerConfi
 
             if (isset($providerConfiguration['requestPatterns']) && is_array($providerConfiguration['requestPatterns'])) {
                 $requestPatterns = array();
-                foreach ($providerConfiguration['requestPatterns'] as $patternType => $patternConfiguration) {
+                foreach ($providerConfiguration['requestPatterns'] as $patternName => $patternConfiguration) {
+                    if (is_string($patternConfiguration)) {
+                        $patternType = $patternName;
```

Comment:
I'd love a small comment explaining which parts of the code are needed for backwards compatibility; i.e. this one ;)